lighthouse-core/gather/driver.js

```diff
@@ -512,6 +505,52 @@ class Driver {
     });
   }
 
+  /**
+   * Rejects if the security state is insecure.
+   * @return {{promise: Promise<void>, cancel: function(): void}}
+   * @private
+   */
+  _waitForSecurityCheck() {
+    /** @type {(() => void)} */
+    let cancel = () => {
+      throw new Error('_waitForSecurityCheck.cancel() called before it was defined');
+    };
+
+    const promise = new Promise(async (resolve, reject) => {
+      /**
+       * @param {LH.Crdp.Security.SecurityStateChangedEvent} event
+       */
+      const securityStateChangedListener = ({securityState, explanations}) => {
+        if (securityState === 'insecure') {
+          cancel();
+          const insecureDescriptions = explanations
+            .filter(exp => exp.securityState === 'insecure')
+            .map(exp => exp.description);
+          const err = new LHError(LHError.errors.INSECURE_DOCUMENT_REQUEST, {
+            securityMessages: insecureDescriptions.join(' '),
+          });
+          reject(err);
+        } else if (securityState !== 'neutral') {
+          cancel();
+          resolve();
+        }
+      };
+
+      this.on('Security.securityStateChanged', securityStateChangedListener);
+      this.sendCommand('Security.enable');
```

Comment:
Confirmed this will crash with `Unhandled Rejection. Reason: LHError: PROTOCOL_TIMEOUT`

Reply:
it's k now